### ISSUE TITLE
Reaction on primary key compromise

### ISSUE THREAD

**Issue by roeslpa** [Qabel/qabel.github.io#78]
If we discuss about forward secrecy, we also need to define a reaction on the compromise of a primary key.
IMO it could be a signed broadcast message to all contacts with a short "I'm compromised" content. Afterwards the contacts delete the primary keys of the contact and wait for a new personal key exchange.

**Comment by schulze** [comment]
This sounds like a good idea and definitely is POST BETA. But we first should think this through properly, as it has at least implications on the user interface, which shouldn't mention private keys.

There also might be other gotchas we are not aware of. I haven't thought much about key revocation yet and don't know how this is handled in other systems (expect OpenPGP).

**Comment by thechauffeur** [comment]
I agree that revocation is POST BETA. For the BETA the only thing you can do is creating a new identity and manually tell your contacts.

Maybe we can implement something like creating a new key and automatically telling the contacts wit the "old" identity (which is technically a new one after this because the key changed).

**Comment by roeslpa** [comment]
As soon as the old key is compromised you should not use it to distribute a new key since an attacker who owns the old key could do the same..

**Comment by audax** [comment]
By using the register server which verifies the email address, we can at least give the user the chance for a new key exchange by just uploading his new key.

**Comment by roeslpa** [comment]
As far as this/every upload is verified by an email, this is an option for users who use the register server.

**Comment by roeslpa** [comment]
In contrast to https://github.com/Qabel/qabel-core/issues/632 this shall also work for identities which are not listed in the index and maybe are still in possession of their identity.
